File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/ThicknessContentTag.java
package com.mcreater.fxfluent.xaml.tags;

import javafx.geometry.Insets;

public class ThicknessContentTag extends SimpleContentTag<Insets> {

    public Insets toObject() {
        String s = element.getData().toString();
        try {
            return new Insets(Integer.parseInt(s));
        }
        catch (Exception ignored) {}

        try {
            String[] sl = s.split(",");
            return new Insets(Integer.parseInt(sl[0]), Integer.parseInt(sl[1]), Integer.parseInt(sl[2]), Integer.parseInt(sl[3]));
        }
        catch (Exception ignored) {}
        return Insets.EMPTY;
    }

    public String toString() {
        return String.format("Insets(%.2f, %.2f, %.2f, %.2f)", toObject().getTop(), toObject().getRight(), toObject().getBottom(), toObject().getLeft());
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/ThicknessContentTag.kt
package com.mcreater.fxfluent.xaml.tags

import javafx.geometry.Insets

class ThicknessContentTag : SimpleContentTag<Insets>() {
    override fun toObject(): Insets {
        val s = element?.data.toString()
        try {
            return Insets(s.toInt().toDouble())
        } catch (ignored: Exception) {
        }

        try {
            val sl = s.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            return Insets(
                sl[0].toInt().toDouble(),
                sl[1].toInt().toDouble(),
                sl[2].toInt().toDouble(),
                sl[3].toInt().toDouble()
            )
        } catch (ignored: Exception) {
        }
        return Insets.EMPTY
    }

    override fun toString(): String {
        return String.format(
            "Insets(%.2f, %.2f, %.2f, %.2f)",
            toObject().top,
            toObject().right,
            toObject().bottom,
            toObject().left
        )
    }
}